src/xo-app/host/tab-network.js

```diff
@@ -117,57 +128,143 @@ class PifItem extends Component {
+  }
+}
+
+class PifItemMode extends Component {
+  state = { configModes: [] }
+
+  componentDidMount () {
+    getIpv4ConfigModes().then(configModes =>
+      this.setState({ configModes })
+    )
+  }
+
+  _configIp = mode => {
+    if (mode === 'Static') {
+      return confirm({
+        icon: 'ip',
+        title: _('pifConfigureIp'),
+        body: <ConfigureIpModal pif={this.props.pif} />
+      }).then(
+        params => {
+          if (!params.ip || !params.netmask) {
+            error(_('configIpErrorTitle'), _('configIpErrorMessage'))
+            return
+          }
+          return reconfigurePifIp(this.props.pif, { mode, ...params })
+        },
+        noop
+      )
+    }
+    return reconfigurePifIp(this.props.pif, { mode })
+  }
 
   render () {
-    const { networks, pif, vifsByNetwork } = this.props
+    const {pif} = this.props
     const { configModes } = this.state
+    return <Select
+      onChange={this._configIp}
+      options={configModes}
+      value={pif.mode}
+    >
+      {pif.mode}
+    </Select>
+  }
+}
 
+@connectStore(() => ({
+  vifsByNetwork: createGetObjectsOfType('VIF').groupBy('$network')
+}))
+class PifItemInUse extends Component {
+  render () {
+    const {networks, pif, vifsByNetwork} = this.props
     const pifInUse = some(vifsByNetwork[pif.$network], vif => vif.attached)
+    return <div className='text-xs-center'>
+      {_toggleDefaultLockingMode(
+        <Toggle
+          disabled={pifInUse}
+          onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
+          value={networks[pif.$network].defaultIsLocked}
+        />,
+        pifInUse && _('pifInUse')
+      )}
+    </div>
+  }
+}
 
-    return <tr>
-      <td>{pif.device}</td>
-      <td>{networks[pif.$network].name_label}</td>
-      <td>
-        {pif.vlan === -1
-          ? 'None'
-          : <Number value={pif.vlan} onChange={this._editPif}>
-            {pif.vlan}
-          </Number>
-        }
-      </td>
-      <td>
-        {pif.ip}
-        {' '}
-        {pif.ip && <a className='hidden-md-down' onClick={this._onEditIp} style={EDIT_BUTTON_STYLE}>
-          <Icon icon='edit' size='1' fixedWidth />
-        </a>}
-      </td>
-      <td>
-        <Select
-          onChange={this._configIp}
-          options={configModes}
-          value={pif.mode}
-        >
-          {pif.mode}
-        </Select>
-      </td>
-      <td><pre>{pif.mac}</pre></td>
-      <td>{pif.mtu}</td>
-      <td className='text-xs-center'>
-        {_toggleDefaultLockingMode(
-          <Toggle
-            disabled={pifInUse}
-            onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
-            value={networks[pif.$network].defaultIsLocked}
-          />,
-          pifInUse && _('pifInUse')
-        )}
-      </td>
-      <td>
+const COLUMNS = [
+  {
+    itemRenderer: (pif, networks) => {
+      return pif.device
+    },
```

Comment:
You don't need the brackets and the `return`. Same thing for all the other `itemRenderer`s that directly return a value.